packages/aws-cdk-lib/aws-appsync/lib/appsync-function.ts

```diff
@@ -152,6 +167,10 @@ export class AppsyncFunction extends Resource implements IAppsyncFunction {
       throw new Error('Mapping templates cannot be used alongside code');
     }
 
+    if (props.maxBatchSize && !(props.dataSource instanceof LambdaDataSource)) {
```

Comment:
for my own understanding, what are the other datatypes and why can't maxBatchSize be set for them?

Reply:
Other data types such as DynamoDB can also be used.
https://docs.aws.amazon.com/appsync/latest/devguide/data-source-components.html

The reason it can only be used with Lambda is that the parameter is used with BatchInvoke when Lambda is the data source. When I actually tried it with other data sources, it resulted in an error.
https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-appsync-functionconfiguration.html#cfn-appsync-functionconfiguration-maxbatchsize

> The maximum number of resolver request inputs that will be sent to a single AWS Lambda function in a BatchInvoke operation.

The following is an actual use case.
https://docs.aws.amazon.com/appsync/latest/devguide/tutorial-lambda-resolvers.html#advanced-use-case-batching